src/read/elf/file.rs

```diff
@@ -297,8 +297,29 @@ where
         match self.header.e_type(self.endian) {
             elf::ET_REL => ObjectKind::Relocatable,
             elf::ET_EXEC => ObjectKind::Executable,
-            // TODO: check for `DF_1_PIE`?
-            elf::ET_DYN => ObjectKind::Dynamic,
+            elf::ET_DYN => {
+                let mut is_pie = false;
+                let table: &SectionTable<'_, Elf, R> = self.elf_section_table();
+                if let Ok(Some(dyn_sec)) = table.dynamic(self.endian(), self.data()) {
+                    for v in dyn_sec.0 {
+                        let tag = v.tag32(self.endian());
+                        let val = v.val32(self.endian());
+                        if let Some(tag) = tag {
+                            if let Some(val) = val {
+                                if tag == DT_FLAGS_1 && val & DF_1_PIE == DF_1_PIE {
+                                    is_pie = true;
+                                }
+                            }
+                        }
+                    }
+                }
+
+                if is_pie {
+                    ObjectKind::PIE
+                } else {
+                    ObjectKind::Dynamic
+                }
+            }
```

Comment:
Can you explain more about how you intend to use this, or even better give a link to the code where you use this. Do you actually need `PIE`, or would returning `Executable` suffice?

The question mark on the TODO was because I wasn't sure this is something we actually want. We also need to be careful because existing users may not want this change, so we'll need to review those.

Reply:
Ok. Actually I need this for addr2line crate of yours

https://github.com/AFLplusplus/LibAFL/blob/main/libafl_qemu/src/modules/utils/addr2line.rs#L111
this is the example.
to look up an address from the `Loader` struct in addr2line. I need to check for if the binary is PIE executable or not.
1) if it is PIE executable, i need to use the relative address
(for example, if your binary is loaded at 0x555555556000 and you want to look for 0x555555556123, then I need to use 0x123 to lookup from the loader)
2) on the other hand, if it is not a PIE executable, I can just directly use the addres where it was loaded.

To detect this difference is necessary for me to use addr2line

Reply:
So case 1 is `ET_DYN`+`DF_1_PIE`, and case 2 is `ET_EXEC`. How do you need to handle `ET_DYN` without `DF_1_PIE`?

Reply:
My understanding is that case 1 is how `ET_DYN` should always be handled (whether it has `DF_1_PIE` or not), so you don't actually need this change. Based on that, I'm going to close this PR, but please discuss further if your needs are not met.